File: c++/1_Two_Sum.cpp
#include <unordered_map>
#include <vector>

using namespace std;

class TwoSum {
public:
    std::vector<int> twoSum(std::vector<int>& nums, int target) {
        std::unordered_map <int, int> hashmap;
        std::vector<int> pair;
        for (int i = 0; i < nums.size(); i++){
            if (hashmap.contains(nums[i])){
                pair = {i, hashmap[nums[i]]};
                return pair;
            }
            hashmap[target-nums[i]] = i;
        }
        return pair;
    }
};

File: python/1_Two_Sum.py
class Two_Sum:
  def twoSum(self, nums: list[int], target: int) -> list[int]:
    myHash = dict()
    pair = list()
    for indx, val in enumerate(nums):
      if (val in myHash):
        indx2 = myHash[val]
        pair = (indx, indx2)
        return pair
      myHash[target-val] = indx